### PR TITLE
added github workflows; added dockerfile; added docker-compose;

### PR DESCRIPTION
**Pull request by skluthe**
Currently having an issue running the container:

```
calendoso_1  | internal/modules/cjs/loader.js:883
calendoso_1  |   throw err;
calendoso_1  |   ^
calendoso_1  | 
calendoso_1  | Error: Cannot find module '/root/.npm/_npx/240/lib/node_modules/prisma/scripts/preinstall-entry.js'
calendoso_1  |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
calendoso_1  |     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
calendoso_1  |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
calendoso_1  |     at internal/main/run_main_module.js:17:47 {
calendoso_1  |   code: 'MODULE_NOT_FOUND',
calendoso_1  |   requireStack: []
calendoso_1  | }
calendoso_1  | npm ERR! code ELIFECYCLE
calendoso_1  | npm ERR! errno 1
calendoso_1  | npm ERR! prisma@2.21.2 preinstall: `node scripts/preinstall-entry.js`
calendoso_1  | npm ERR! Exit status 1
calendoso_1  | npm ERR! 
calendoso_1  | npm ERR! Failed at the prisma@2.21.2 preinstall script.
calendoso_1  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
calendoso_1  | 
calendoso_1  | npm ERR! A complete log of this run can be found in:
calendoso_1  | npm ERR!     /root/.npm/_logs/2021-04-15T15_49_11_857Z-debug.log
calendoso_1  | Install for [ 'prisma@latest' ] failed with code 1
```
But other than that it should be pretty good. The workflows require you to add a PAT with the packages scope from github and also a dockerhub username and password to the secrets for this repo. 

It'll automatically build for multiarch (arm, x86, etc) and push to ghcr.io and dockerhub. I also added a docker-compose.yml for testing. 

If you have any questions, a solution for the error above, or any requests to change this PR, feel free to let me know. Most of it was copied from https://github.com/selfhostedpro/yacht if you're curious about how any of this works that may give you some insight.

### CODE REVIEW

**Comment by skluthe** [comment]
Didn't see #7 sorry about that! May be some useful stuff from this to copy over.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: "3.1"
+
+services:
+  db:
+    image: postgres
+    restart: always
+    environment:
+      POSTGRES_PASSWORD: ${DB_PASS:-changeme}
+
+  calendoso:
```
Comment (50bbx) [inline docker-compose.yml:10]:
```suggestion
  calendso:
```

**Comment by fabioelia** [comment]
@skluthe ran into a similar issue, it looks like the `postinstall` https://github.com/calendso/calendso/blob/main/package.json#L9 script isn't running. I had to run `yarn postinstall` manually before the primsa command in order for it to run

**Comment by baileypumfleet** [comment]
Going to close this PR as we're not officially supporting Docker. We may reconsider supporting it in future. Thanks very much for your help though!

**Comment by PeerRich** [comment]
docker support now exists here: https://github.com/calendso/calendso-docker powered by the community. not officially maintained by the calendso core team, yet.

**Pull request closed without merging**